enumerables.rb

```diff
@@ -0,0 +1,101 @@
+module Enumerable
+
+    def my_each
+        for i in 0..self.length-1
+            yield(self[i])
+        end
+    end
+    
+    def my_each_with_index
+        for i in 0..self.length-1
+            yield(self[i],i)
+        end
+    end
+
+    def my_select
+        arr = []
+        for i in 1..self.length
+            if yield(i)
+                arr.push(i)
+            end
+        end
+        arr
+    end
+
+    def my_all?
+        results = true
+        for i in 0..self.length-1
+            if yield(self[i])
+                results = true
+            end
+        end
+        results
+    end
+
+    def my_any?
+        results = true
+        for i in 0..self.length-1
+            if yield(self[i])
+                results = true
+            end
+        end
+        results
+    end
+
+    def my_none?
+        results = true
+        for i in 0..self.length-1
+            if yield(self[i])
+                results = false
+            end
+        end
+        results
+    end
+
+    def my_count
+        counter = 0
+        for i in 0..self.length-1
+            if yield(self[i])
+                counter +=1
+            end
+        end
+        counter
+    end
```

Comment:
* [ ] my_count returns the number of items in enum through enumeration
  ```ruby
    array = [1,2,5,4,7]
    p array.my_count == array.count # returns this error
      #
      # LocalJumpError:
        no block given (yield)
      # ./enumerables.rb:58:in `block in my_count'
      # ./enumerables.rb:57:in `each'
      # ./enumerables.rb:57:in `my_count'
  ```

* [ ] my_count counts the number of items in enum that are equal to item if an argument is given
```ruby
array = [7,8,9,5,1,0,2,4,0,1]
array.my_count(0) == array.count(0) # false
```